packages/persist/lib/routes/environment/environmentId/connection/connectionId/sync/syncId/job/jobId/deleteRecords.ts

```diff
@@ -0,0 +1,63 @@
+import type { ApiError, Endpoint } from '@nangohq/types';
+import type { EndpointRequest, EndpointResponse, RouteHandler, Route } from '@nangohq/utils';
+import { persistRecords, recordsPath } from '../../../../../../../../../records.js';
+import { validateRecords } from './validate.js';
+
+type DeleteRecords = Endpoint<{
+    Method: typeof method;
+    Path: typeof path;
+    Params: {
+        environmentId: number;
+        nangoConnectionId: number;
+        syncId: string;
+        syncJobId: number;
+    };
+    Body: {
+        model: string;
+        records: Record<string, any>[];
+        providerConfigKey: string;
+        connectionId: string;
+        activityLogId: string;
+    };
+    Error: ApiError<'delete_records_failed'>;
+    Success: never;
+}>;
+
+const path = recordsPath;
+const method = 'DELETE';
+
+const validate = validateRecords<DeleteRecords>();
+
+const handler = async (req: EndpointRequest<DeleteRecords>, res: EndpointResponse<DeleteRecords>) => {
+    const {
+        params: { environmentId, nangoConnectionId, syncId, syncJobId },
+        body: { model, records, providerConfigKey, connectionId, activityLogId }
+    } = req;
+    const result = await persistRecords({
+        persistType: 'delete',
+        environmentId,
+        connectionId,
+        providerConfigKey,
+        nangoConnectionId,
+        syncId,
+        syncJobId,
+        model,
+        records,
+        activityLogId
+    });
+    if (result.isOk()) {
+        res.status(201).send();
```

Comment:
201 is incorrect imo, 204 if you want
(for consistency I'm always leaning toward 200 with content-type json but as you want)

Reply:
thanks. I always get 201 and 204 confused, I don't know why